src/components/ChatList/GroupChatChannelItem.tsx

```diff
@@ -0,0 +1,64 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+/* eslint-disable no-use-before-define */
+import React from 'react';
+import {View} from 'react-native';
+
+import ChannelContent from './elements/ChannelContent';
+import ChannelImage from './elements/ChannelImage';
+import CustomPressable from '../CustomPressable';
+import useProfileHook from '../../hooks/core/profile/useProfileHook';
+import {ChannelItemProps} from '../../../types/component/ChatList/ChannelItem.types';
+import {calculateTime} from '../../utils/time';
+import {channelItemStyles as styles} from './ChannelItem.style';
+
+const GroupChatChannelItem = (props: ChannelItemProps) => {
+  const {channel: groupChat, onChannelPressed} = props;
+  const {signedProfileId} = useProfileHook();
+  const channelType = 'GROUP';
+  const channelPicture = groupChat?.channelPicture;
+
+  const unreadCount = groupChat?.unreadCount;
+  const hasBadge = unreadCount > 0;
+
+  const description = groupChat?.rawJson?.firstMessage ?? groupChat?.rawJson?.message;
+  const isSystemDescription = description?.isSystem ?? description?.type === 'system';
+  const isDeletedMessage = description?.type === 'deleted';
+  const sender = description?.user?.username ?? description?.user?.name;
+  const isMeAsSender = description?.user?.id === signedProfileId;
+
+  let channelDescription = groupChat?.description;
+  if (!isSystemDescription) {
+    channelDescription = isMeAsSender
+      ? `You: ${channelDescription}`
+      : `${sender}: ${channelDescription}`;
+  }
+  if (!sender || isDeletedMessage) channelDescription = '';
+
+  return (
+    <CustomPressable onPress={onChannelPressed}>
+      <View style={styles.channelContainer}>
+        <ChannelImage>
+          <ChannelImage.Big type={channelType} image={channelPicture} />
+          <ChannelImage.Small type={channelType} />
+        </ChannelImage>
+
+        <View style={styles.contentContainer}>
+          <ChannelContent>
+            <View style={{flexDirection: 'row'}}>
+              <ChannelContent.Title>{groupChat?.name}</ChannelContent.Title>
+              <ChannelContent.Time>
+                {calculateTime(groupChat?.lastUpdatedAt, true)}
+              </ChannelContent.Time>
+            </View>
+            <View style={{flexDirection: 'row'}}>
+              <ChannelContent.Description>{channelDescription}</ChannelContent.Description>
+              {hasBadge && <ChannelContent.Badge>{unreadCount}</ChannelContent.Badge>}
+            </View>
+          </ChannelContent>
+        </View>
+      </View>
+    </CustomPressable>
+  );
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `GroupChatChannelItem` component seems to be well implemented. However, there is a potential issue with the way you're handling the channel description. If the message is a system message or a deleted message, you're setting the channel description to an empty string. This might not be the best user experience as it could lead to confusion. Consider providing a more descriptive placeholder for these cases.

```diff
-  if (!sender || isDeletedMessage) channelDescription = '';
+  if (!sender || isDeletedMessage) channelDescription = 'Message unavailable';
```

Also, consider extracting some of the logic into separate functions to improve readability and maintainability. For example, the logic for determining the channel description could be extracted into a function like `getChannelDescription`.

```diff
-  let channelDescription = groupChat?.description;
-  if (!isSystemDescription) {
-    channelDescription = isMeAsSender
-      ? `You: ${channelDescription}`
-      : `${sender}: ${channelDescription}`;
-  }
-  if (!sender || isDeletedMessage) channelDescription = '';

+  const channelDescription = getChannelDescription(groupChat, isSystemDescription, isMeAsSender, sender, isDeletedMessage);

...

+ function getChannelDescription(groupChat, isSystemDescription, isMeAsSender, sender, isDeletedMessage) {
+   let description = groupChat?.description;
+   if (!isSystemDescription) {
+     description = isMeAsSender
+       ? `You: ${description}`
+       : `${sender}: ${description}`;
+   }
+   if (!sender || isDeletedMessage) description = 'Message unavailable';
+   return description;
+ }
```
This makes the code easier to read and test, and it also adheres to the Single Responsibility Principle (SRP) by ensuring that each function does one thing only.


<!-- This is an auto-generated comment by OSS CodeRabbit -->